chap_core/rest_api_src/v1/routers/analytics.py

```diff
@@ -324,3 +332,54 @@ class DataSource(DBModel):
 @router.get('/data-sources', response_model=List[DataSource])
 async def get_data_sources() -> List[DataSource]:
     return data_sources
+
+
+@router.post("/create-backtest-with-data", response_model=JobResponse)
+async def create_backtest_with_data(
+    request: MakeBacktestWithDataRequest,
+    database_url: str = Depends(get_database_url),
+    worker_settings=Depends(get_settings),
+):
+    feature_names = list({entry.feature_name for entry in request.provided_data})
+
+    dataclass = create_tsdataclass(feature_names)
+    provided_data_processed = observations_to_dataset(
+        dataclass,
+        request.provided_data,
+        fill_missing=True
+    )
+
+    if "population" in feature_names:
+        provided_data_processed = provided_data_processed.interpolate(["population"])
+
+    for feature_name in feature_names:
+        if feature_name == "disease_cases":
+            continue
+        for location, data in provided_data_processed.items():
+            isnan = np.isnan(getattr(data, feature_name))
+            if np.any(isnan):
+                isnan_ = [data.time_period[i] for i in np.flatnonzero(isnan)]
+                raise HTTPException(
+                    status_code=500,
+                    detail=f"Missing value in {feature_name} in location {location}. Time periods: {isnan_}",
+                )
```

Comment:
I don't fully understand everything that is happening here, so please have an extra look.